src/foam/u2/crunch/CapabilityInterceptView.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.crunch',
+  name: 'CapabilityInterceptView',
+  extends: 'foam.u2.View',
+
+  implements: [ 'foam.mlang.Expressions' ],
+
+  requires: [
+    'foam.u2.layout.Grid',
+    'foam.u2.layout.GUnit',
+    'foam.u2.crunch.CapabilityCardView',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction'
+  ],
+
+  imports: [
+    'capabilityAquired',
+    'capabilityCancelled',
+    'crunchController',
+    'stack',
+    'capabilityCache',
+    'user',
+    'userCapabilityJunctionDAO'
+  ],
+
+  properties: [
+    {
+      name: 'capabilityOptions',
+      class: 'StringArray'
+    },
+    {
+      name: 'capabilityView',
+      class: 'foam.u2.ViewSpec',
+      factory: function () {
+        return 'foam.u2.crunch.CapabilityCardView';
+      }
+    }
+  ],
+
+  css: `
+    ^ {
+      width: 1024px;
+      margin: auto;
+    }
+    ^detail-container {
+      overflow-x: scroll;
+    }
+  `,
+
+  methods: [
+    function initE() {
+      this.capabilityOptions.forEach((c) => {
+        if ( this.capabilityCache.has(c) && this.capabilityCache.get(c) === true ) {
+          capabilityAquired = true;
+          this.stack.back();
```

Comment:
thinking you have a good suggestion here. Will work through this and update soon.

Reply:
Ok so thought about this a little more:
#### Situation:
User selects an action and finds they don't have permission to do this, but there is a capability that will grant this permission. 
#### Current Distinction:
Right now this permission check only goes one level in. An auth check is done to see if the user has this **one** permission and the capabilityAuthService returns the list of capabilities that will grant this.
#### Purposed Changes - from my understanding of Patrick's comment:
The permission check goes through all possible exceptions and builds list of permissions needed, then when we call the interceptView we have potentially multiple capabilities that need to be completed.
However the issues with this come from potentially needing the user to select which capability they want to complete, since their can be different options. So after they complete one capability for one permission they may decide on other capability for other permissions.

I would suggest that despite being an interesting idea, and due to the complications as well as the fact that users should be able to select from multiple capabilities for any specific permission - makes me believe this flow makes sense as is.

Reply:
I was under the impression this logic was used to display all the `required` capabilities in order to satisfy a permission but as @Nauna described it's to provide a list of all - let me clarify this - top level capabilities that grant the permission.

Seems odd considering that if the user has a capability that grants them the permission that caused the exception in the first place, how does the exception get thrown if they have it granted?

Reply:
An exception won't be thrown if the capability is granted since it gets thrown by CapabilityAuthService. If there were two capabilities displayed and the user wants to get both of them then they would currently need to go to the Capability Store to get the second one.

Reply:
Why would those two capabilities be displayed in the first place? If the user has the permission through a capability that's been granted to them then the exception wouldn't be thrown and this logic wouldn't be necessary.

From what I understand the capabilityAuthService is responsible for checking if the user has the permission through its capabilities and throws an exception depending on the result, which the controller then catches and displays the intercept view which then checks for the permission again. Unless I'm missing something then there's a bit of redundancy going on.

Reply:
An action that checks the auth service, sees a permission is missing, and then checks to see if any capability can grant this permission. The list of potential capabilities is given to the InterceptView which displays these capabilities to the user. If the user completes one and that capability is granted, the intercept pushes the user back to the original action. However if the capability is not granted, the user is redirected back to the intercept where they have the option to select another capability - if they so choose.

Note that a capability can go into a pending state where the user must wait for some other service to complete (ex KYC for a signing officer).

Not seeing a redundancy?

Reply:
It is also possible that the original action has another permission that is needed, and may call another list of capabilities.